File: app/src/main/java/com/hjc/reader/ui/wan/adapter/MyBannerAdapter.java
package com.hjc.reader.ui.wan.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.hjc.reader.R;
import com.hjc.reader.bean.response.WanBannerBean;
import com.hjc.reader.databinding.ItemBannerBinding;
import com.youth.banner.adapter.BannerAdapter;

import java.util.List;

public class MyBannerAdapter extends BannerAdapter<WanBannerBean.DataBean, MyBannerAdapter.BannerViewHolder> {

    public MyBannerAdapter(List<WanBannerBean.DataBean> data) {
        super(data);
    }


    @Override
    public BannerViewHolder onCreateHolder(ViewGroup parent, int viewType) {
        ItemBannerBinding viewDataBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_banner, parent, false);
        return new BannerViewHolder(viewDataBinding);
    }

    @Override
    public void onBindView(BannerViewHolder holder, WanBannerBean.DataBean data, int position, int size) {
        if (data == null){
            return;
        }
        if (holder.mBinding != null){
            holder.mBinding.setBannerBean(data);

            String num = (position + 1) + "/" + size;
            holder.mBinding.tvNum.setText(num);
        }
    }

    static class BannerViewHolder extends RecyclerView.ViewHolder {
        ItemBannerBinding mBinding;

        BannerViewHolder(ItemBannerBinding binding) {
            super(binding.getRoot());
            this.mBinding = binding;
        }
    }
}

File: module-wan/src/main/java/com/hjc/learn/wan/adapter/MyBannerAdapter.kt
package com.hjc.learn.wan.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.hjc.learn.wan.R
import com.hjc.learn.wan.databinding.WanItemBannerBinding
import com.hjc.library_net.model.WanBannerBean
import com.youth.banner.adapter.BannerAdapter

/**
 * @Author: HJC
 * @Date: 2021/2/22 16:40
 * @Description: banner adapter
 */
class MyBannerAdapter(data: MutableList<WanBannerBean>?) :
    BannerAdapter<WanBannerBean, MyBannerAdapter.BannerViewHolder>(data) {

    override fun onCreateHolder(parent: ViewGroup, viewType: Int): BannerViewHolder {
        val viewDataBinding: WanItemBannerBinding =
            DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.wan_item_banner, parent, false)
        return BannerViewHolder(viewDataBinding)
    }

    override fun onBindView(holder: BannerViewHolder, data: WanBannerBean?, position: Int, size: Int) {
        if (data == null) {
            return
        }

        holder.mBinding?.let {
            it.bannerBean = data
            val num = (position + 1).toString() + "/" + size
            it.tvNum.text = num
        }
    }

    class BannerViewHolder(binding: WanItemBannerBinding) : RecyclerView.ViewHolder(binding.root) {
        var mBinding: WanItemBannerBinding? = binding
    }

}